onnxruntime/core/providers/mkldnn/activation/activations.cc

```diff
@@ -0,0 +1,204 @@
+    
+    mkldnn::memory::format fmt = mkldnn::memory::format::any;
+    switch (params.num_dimensions) {
+    case 1: { fmt = mkldnn::memory::format::x; break; }
+    case 2: { fmt = mkldnn::memory::format::nc; break; }
+    case 3: { fmt = mkldnn::memory::format::ntc; break; }
+    case 4: { fmt = mkldnn::memory::format::nchw; break; }
+    case 5: { fmt = mkldnn::memory::format::ncdhw; break; }
+    default: {  fmt = mkldnn::memory::format::any; break; }
+    }
+
+    context_.src_md.reset(new mkldnn::memory::desc({ params.src_dims}, MklDnnType<T>(), fmt));
+
+    mkldnn::algorithm algo = mkldnn::algorithm::eltwise_relu;
+    context_.fwd_desc.reset(new mkldnn::eltwise_forward::desc(
+      mkldnn::prop_kind::forward_inference, algo, *context_.src_md, 0));
+
+    context_.relu_fwd_pd.reset(new mkldnn::eltwise_forward::primitive_desc(
+      *context_.fwd_desc, cpu_engine_));
+
+    context_.src_fmt = static_cast<mkldnn::memory::format>(
+      context_.relu_fwd_pd.get()->src_primitive_desc().desc().data.format);
+
+    context_.src_size = context_.relu_fwd_pd.get()->src_primitive_desc().get_size();
+    context_.dst_size = context_.relu_fwd_pd.get()->dst_primitive_desc().get_size();
+
+    context_.src_mem.reset(new mkldnn::memory(context_.relu_fwd_pd.get()->src_primitive_desc(), nullptr));
+    context_.dst_mem.reset(new mkldnn::memory(context_.relu_fwd_pd.get()->dst_primitive_desc(), nullptr));
+    context_.relu_fwd.reset(
+      new mkldnn::eltwise_forward(*context_.relu_fwd_pd, *context_.src_mem, *context_.dst_mem));
+    context_.net.push_back(*context_.relu_fwd);
+  }
+
+  ReluContext context_;
+  mkldnn::engine& cpu_engine_;
+};
+
+// Pool which allows for reuse of MKLDNN Relu primitives which are expensive 
+// to instantiate. To address thread safety, the primitives are stored in a map 
+// on thread local storage.
+template <typename T>
+class ReluPrimitivePool : public PrimitivePool<T> {
+ public:
+  static ReluPrimitive<T>* Get(const ReluParams& params) {
+    ReluPrimitive<T>* primitive = dynamic_cast<ReluPrimitive<T>*>(
+        ReluPrimitivePool<T>::GetInstance().GetPrimitive(params.ToString()));
+
+    if (primitive == nullptr) {
+      auto relu_primitive = std::make_unique<ReluPrimitive<T>>(params);
+      primitive = relu_primitive.get();
+      ReluPrimitivePool<T>::GetInstance().SetPrimitive(params.ToString(), 
+		std::move(relu_primitive));
+    }
+    return primitive;
+  }
+
+ private:
+  ReluPrimitivePool() = default;
+  ~ReluPrimitivePool() = default;
+
+  static ReluPrimitivePool& GetInstance() {
+    static ReluPrimitivePool pool;
+    return pool;
+  }
+};
+}	// namespace
+
+template <typename T>
+Status Relu<T>::Compute(OpKernelContext* context) const {
+  const Tensor* X = context->Input<Tensor>(0);
+  Tensor* Y = context->Output(0, X->Shape());
+  
+  const TensorShape& x_shape = X->Shape();
+  const auto& x_dims = x_shape.GetDims();
+
+  if (X->Shape().NumDimensions() > 5 ) {
```

Comment:
can you leave a comment here saying why we're defaulting to the non-mkldnn implementation?